oeps/oep-0021-proc-deprecation.rst

```diff
@@ -0,0 +1,310 @@
+
+.. _open-edx-proposals#63: https://github.com/edx/open-edx-proposals/pull/63
+
+Abstract
+========
+
+Open edX needs a formal process on how to deprecate and remove unused or
+unneeded technology.
+
+This document details that process.
+
+Motivation
+==========
+
+There currently is no formal process on how to identify and deprecate
+technology that can be removed from the Open edX platform. As a result, the
+state of technologies that can be removed or are already partially
+removed is captured in a non-standardized set of formats. This document outlines
+common language and a process to use when making the decision on what to
+deprecate, when it should be done, and the status of execution. The result of
+following this process should be alignment and the right degree of involvement
+from all stakeholders.
+
+Specification
+==============
+
+When a team or individual believes that any technology on the Open edX
+platform should no longer be used or maintained, the following steps should be
+followed.
+
+Criteria and identification
+---------------------------
+
+The list of technology that should be considered for depreciation and
+removal from the edX platform includes but is not limited to:
+
+- REST APIs
+
+- Apps
+
+- Learner or partner facing features (user interfaces and associated backend)
+
+- Backend features - infrastructure and integrations
+
+- Feature toggles (multiple code paths)
+
+A general rule to follow is that if it’s code that's used by any Open edX
+platform feature then the associated deprecation/removal should be handled by
+the process outlined in this OEP.
+
+Once the technology to be deprecated is identified, the originator
+should be prepared to communicate the following:
+
+- Short description of what the technology is
+
+- Where it's currently in use (i.e. repos) within the Open edX codebase
+
+- Who uses it (e.g. learners, course authors, etc.)
+
+- Target deadlines for deprecation and removal
+
+Deprecation steps
+-----------------
+
+1. Identify the technology to be deprecated
+
+2. Communicate out to edX staff and the Open edX community in accordance with
+the communication section below.
+
+3. Create a JIRA ticket of issue type [Task] with the label 
+[deprecation-removal]. After
+creation, confirm that the ticket shows up in the Deprecation/Removal JIRA
+`board <https://openedx.atlassian.net/secure/RapidBoard.jspa?rapidView=452>`_
+
+4. Initiate discussion as needed with the relevant edX team(s) that own the
+technology or other stakeholders (e.g. consumers of the technology) on intention
+and timeline for removal.
+
+5. Update the ticket with relevant information (see section above for suggested
+details) and anticipated timeline. See [Suggested-Timeline] section for
+suggestions on possible timelines. Continue to update the ticket description
+with stakeholder feedback as applicable and update to "Accepted" once it's been
+decided that there's alignment. 
```

Comment:
Please double-check that deprecation tickets are open to community members. Or, if for some reason it cannot be, make sure that any stakeholders can watch the ticket.

Reply:
Makes sense, I'll be checking with edX IT along the same lines as with the first comment above.